### PR TITLE
Extends `@SuppressFBWarnings` to allow regex and exact matches

### PR DESCRIPTION
**Pull request by gtoison**
Extend the bug suppression, allowing exact (not prefix) bug type match, and regex bug type match

### CODE REVIEW

File: spotbugs-annotations/src/main/java/edu/umd/cs/findbugs/annotations/SuppressMatchType.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+     */
+    DEFAULT,
+
+    /**
+     * Suppress bugs matching the exact (case sensitive) given type.
+     */
+    EXACT,
+
+    /**
+     * Suppress bug whose type match the given regular expression.
```
Comment (JuditKnoll) [inline spotbugs-annotations/src/main/java/edu/umd/cs/findbugs/annotations/SuppressMatchType.java:19]: IMO it's rather connected to the `value` field that to which properties is it connected to (type, category and abbreviation), so even if it's regex or exact match, both the type, the category and the abbreviation could be considered. Or if not, the comment at the `SuppressFBWarnings.value` to should show this change.

**Review by JuditKnoll — CHANGES_REQUESTED** [review]
Spotless didn't like this PR, can you please run `gradlew spotlessApply`?
Can you please add a few tests for this?
Before merging this to master, it would be great to update the documentation about this as well.

File: spotbugs-annotations/src/main/java/edu/umd/cs/findbugs/annotations/SuppressMatchType.java (hunk 1 above)
Comment (gtoison) [reply spotbugs-annotations/src/main/java/edu/umd/cs/findbugs/annotations/SuppressMatchType.java]: I guess what we're aiming for here is the least surprise principle, so I _think_ it's going to be less surprising if all matchTypes use the same criteria (type, category, abbreviation)
I've changed the code to match either of the 3 for all match types

**Review by JuditKnoll — CHANGES_REQUESTED** [review]

**Review by JuditKnoll — APPROVED** [review]
It might be a good idea to add some info about this feature in the [documentation](https://github.com/spotbugs/spotbugs/blob/master/docs/annotations.rst#eduumdcsfindbugsannotationssuppressfbwarnings) as well, but otherwise LGTM.

**Comment by gtoison** [comment]
I looked for mentions to `@SuppressFBWarnings` in the documentation but couldn't find any, I guess the javadocs _are_ the docs for now :)

**Comment by JuditKnoll** [comment]
> I looked for mentions to `@SuppressFBWarnings` in the documentation but couldn't find any, I guess the javadocs _are_ the docs for now :)

First I thought so as well. But it differs a bit. All I could find about suppressing warnings is in the [annotations.rst](https://github.com/spotbugs/spotbugs/blob/master/docs/annotations.rst#eduumdcsfindbugsannotationssuppressfbwarnings) file.

**Review by ThrawnCA — APPROVED** [review]

**Pull request merged**